src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -662,6 +664,291 @@ public static async ValueTask<FormulaValue> SortTable(EvalVisitor runner, EvalVi
+            }
+
+            var rowsToOrder = new List<(DValue<RecordValue> row, FormulaValue columnValue)>();
+            var blankRows = new List<DValue<RecordValue>>();
+            foreach (var row in arg0.Rows)
+            {
+                if (row.IsError)
+                {
+                    return row.Error;
+                }
+
+                if (row.IsBlank)
+                {
+                    rowsToOrder.Add((row, row.Blank));
+                }
+                else
+                {
+                    var fieldValue = await row.Value.GetFieldAsync(columnName, runner.CancellationToken).ConfigureAwait(false);
+                    if (fieldValue is ErrorValue ev)
+                    {
+                        return ev;
+                    }
+
+                    rowsToOrder.Add((row, fieldValue));
+                }
+            }
+
+            var sortedList = rowsToOrder.OrderBy(pair =>
+            {
+                if (pair.columnValue is BlankValue)
+                {
+                    // Blanks go to the end
+                    return int.MaxValue;
+                }
+                else
+                {
+                    if (!pair.columnValue.TryGetPrimitiveValue(out var primitiveValue))
+                    {
+                        return int.MaxValue;
+                    }
+
+                    var indexOnTable = orderTableValues.IndexOf(primitiveValue);
+                    if (indexOnTable < 0)
+                    {
+                        // If not found, go to the end before the blanks
+                        return int.MaxValue - 1;
+                    }
+                    else
+                    {
+                        return indexOnTable;
+                    }
+                }
+            });
+
+            var orderedRows = sortedList.Select(pair => pair.row).ToList();
+            return new InMemoryTableValue(irContext, orderedRows);
+        }
+
+        private class FormulaValueComparer : IComparer<FormulaValue>
+        {
+            private EvalVisitor Runner { get; }
+
+            private bool IsAscending { get; }
+
+            public FormulaValueComparer(EvalVisitor runner, bool isAscending)
+            {
+                Runner = runner;
+                IsAscending = isAscending;
+            }
+
+            public int Compare(FormulaValue x, FormulaValue y)
+            {
+                var modifier = this.IsAscending ? 1 : -1;
+                if (x is BlankValue)
+                {
+                    return y is BlankValue ? 0 : 1;
+                }
+
+                if (y is BlankValue)
+                {
+                    return -1;
+                }
+
+                if (x is StringValue sv1 && y is StringValue sv2)
+                {
+#pragma warning disable CA1309 // Use ordinal string comparison
+                    return string.Compare(sv1.Value, sv2.Value, StringComparison.CurrentCulture) * modifier;

```

Comment:
>CurrentCulture [](http://example.com/codeflow?start=81&length=14)

Why CurrentCulture? and not Invariant or Culture from Runtime Config?

Reply:
Great catch, using the culture from the runner.